packages/rust/server/src/models/group.rs

```diff
@@ -0,0 +1,163 @@
+use crate::types::DbPool;
+
+use std::borrow::Borrow;
+
+use crate::models::member::Role;
+use crate::models::user::User;
+use sqlx::postgres::PgRow;
+
+use sqlx::types::chrono::NaiveDateTime;
+use sqlx::types::Uuid;
+use sqlx::Row;
+
+pub struct Group {
+    pub id: i32,
+    pub uuid: Uuid,
+    pub group_name: String,
+    pub created: NaiveDateTime,
+}
+
+#[derive(sqlx::FromRow)]
+pub struct UserIDs {
+    pub user_ids: Vec<String>,
+}
+
+impl From<&PgRow> for Group {
+    fn from(row: &PgRow) -> Self {
+        Group {
+            id: row.get("id"),
+            uuid: row.get("uuid"),
+            group_name: row.get("group_name"),
+            created: row.get("created"),
+        }
+    }
+}
+
+impl Group {
+    pub async fn role(&self, db: &DbPool, user_id: i32) -> Result<Role, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM member WHERE group_id = $1 AND user_id = $2;")
+            .bind(self.id)
+            .bind(user_id)
+            .fetch_one(db)
+            .await?;
+        let role_id: i32 = row.get("role_id");
+        //TODO //? what is the todo? -Gannon
```

Comment:
What's this todo?

Reply:
idk, @clef98 had it in his file and I added the statement "//? what is the todo? -Gannon"